frame/staking-rewards/src/lib.rs

```diff
@@ -461,53 +517,69 @@ pub mod pallet {
-					asset_id,
-					reward_configs: initial_reward_config,
-					end_block,
-					lock,
-				} => {
-					ensure!(
+			// Review [Andy]
+			// this is maybe clearer and we needn't handle the first result
+			return if let RewardRateBasedIncentive {
+				owner,
+				asset_id,
+				reward_configs: initial_reward_config,
+				end_block,
+				lock,
+			} = pool_config {
+
+				let pool_id = RewardPoolCount::<T>::increment()?;
+
+				ensure!(
 						end_block > frame_system::Pallet::<T>::current_block_number(),
 						Error::<T>::EndBlockMustBeInTheFuture
 					);
 
-					let pool_id = RewardPoolCount::<T>::increment()?;
+				// Review [Andy] - ensure we can deposit into the new account for the pool
+				// Would be good to have access to `Balances` for this, but for now.
+				let existential_deposit = 1000u32;
+				T::Assets::can_deposit(asset_id, &Self::pool_account_id(&pool_id), existential_deposit.into(), false)
+					.into_result()?;
 
-					let now_seconds = T::UnixTime::now().as_secs();
+				let now_seconds = T::UnixTime::now().as_secs();
 
-					let rewards = initial_reward_config
-						.into_iter()
-						.map(|(asset_id, amount)| {
-							(asset_id, Reward::from_config(amount, now_seconds))
-						})
-						.try_collect()
-						.expect("No items were added; qed;");
+				let rewards = initial_reward_config
+					.into_iter()
+					.map(|(asset_id, amount)| {
+						(asset_id, Reward::from_config(amount, now_seconds))
+					})
+					.try_collect()
+					.expect("No items were added; qed;");
+
+				// Review [Andy]
+				// maybe this could probably be better moved to the caller and the logic
+				// implemented in a type at configuration as mentioned above
+
+				// One question here is whether we really want to have several pools with the same
+				// asset id
+				RewardPools::<T>::insert(
+					pool_id,
+					RewardPool {
+						owner: owner.clone(),
+						asset_id,
+						rewards,
+						total_shares: T::Balance::zero(),
+						claimed_shares: T::Balance::zero(),
+						end_block,
+						lock,
+					},
+				);
 
-					RewardPools::<T>::insert(
-						pool_id,
-						RewardPool {
-							owner: owner.clone(),
-							asset_id,
-							rewards,
-							total_shares: T::Balance::zero(),
-							claimed_shares: T::Balance::zero(),
-							end_block,
-							lock,
-						},
-					);
-					// TODO (vim): Create the financial NFT collection for the rewards pool
-					Ok((owner, pool_id, end_block))
-				},
-				_ => Err(Error::<T>::UnimplementedRewardPoolConfiguration),
-			}?;
-			Self::deposit_event(Event::<T>::RewardPoolCreated { pool_id, owner, end_block });
-			Ok(pool_id)
+				Self::deposit_event(Event::<T>::RewardPoolCreated { pool_id, owner, end_block });
+				Ok(pool_id)
+			} else {
+				Err(Error::<T>::UnimplementedRewardPoolConfiguration.into())
+			}
 		}
 	}
 
+	// Review [Andy]
+	// Maybe the FinancialNftProtocol trait could be an associated type of the pallet and
+	// decoupling it and allow one to mock it simply (as above)
```

Comment:
all the traits in this file are to be implemented by this pallet and depended on by other pallets